src/main/java/org/truffleruby/core/string/ConvertBytes.java

```diff
@@ -288,7 +288,7 @@ private long stringToLong(int nptr, int[] endptr, int base) {
         return 0;
     }
 
-    @TruffleBoundary
+    @TruffleBoundary(transferToInterpreterOnException = false)
```

Comment:
That's an unrelated change, would be to good to split it in another commit at least.